scripts/ci-lint-dockerfiles.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+FILES=$(find -- * -name Dockerfile)
+while read -r file; do
+  echo "Linting: ${file}"
+  # Configure the linter to only fail for errors. Can be set to: error | warning | info | style | ignore | none
+  docker run --rm -i ghcr.io/hadolint/hadolint:v2.10.0 hadolint --ignore DL3006 --failure-threshold warning - < "${file}"
```

Comment:
Let add this individually in the Dockerfiles just for the places we're using build-args (I think that's every Dockerfile right now). That way we'll pick up on new images added without versions.